tracks/kotlin/exercises/rotational-cipher/mentoring.md

```diff
@@ -0,0 +1,42 @@
+### Reasonable Solutions
+
+```kotlin
+class RotationalCipher(private val key: Int) {
+    init {
+        require(key in 0..26) {
+            "Key should be integer between 0 and 26"
+        }
+    }
+
+    fun encode(input: String): String =
+            input.map(shiftBy(key)).joinToString("")
+
+    // Do not allow private members to access object state.
+    // This allows for easier refactoring to another object
+    // because their state is self contained.
+    private fun shiftBy(key: Int): (Char) -> Char = {
```

Comment:
Use '[`Char#isLetter`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/is-letter.html) instead, which allows you to drop `shouldNotShift` completely.

Reply:
👍